programs/clearing_house/src/controller/insurance.rs

```diff
@@ -429,6 +429,62 @@ pub fn remove_insurance_fund_stake(
     Ok(withdraw_amount)
 }
 
+pub fn admin_remove_insurance_fund_stake(
+    insurance_vault_amount: u64,
+    n_shares: u128,
+    spot_market: &mut SpotMarket,
+    now: i64,
+) -> ClearingHouseResult<u64> {
+    apply_rebase_to_insurance_fund(insurance_vault_amount, spot_market)?;
+
+    let total_if_shares_before = spot_market.total_if_shares;
+    let user_if_shares_before = spot_market.user_if_shares;
+
+    let if_shares_before = total_if_shares_before
+        .checked_sub(user_if_shares_before)
+        .ok_or_else(math_error!())?;
+
+    validate!(
+        if_shares_before >= n_shares,
+        ErrorCode::InsufficientLPTokens,
+        "if_shares_before={} < n_shares={}",
+        if_shares_before,
+        n_shares
+    )?;
+
+    let amount = if_shares_to_vault_amount(
+        n_shares,
+        spot_market.total_if_shares,
+        insurance_vault_amount,
+    )?;
+
+    let withdraw_amount = amount;
```

Comment:
why not just set the withdraw amount immediately instead of setting `amount` intermediately?